src/Microsoft.AspNet.Buffering/SendFileFeatureWrapper.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http.Features;
+
+namespace Microsoft.AspNet.Buffering
+{
+    internal class SendFileFeatureWrapper : IHttpSendFileFeature
+    {
+        private readonly IHttpSendFileFeature _originalSendFileFeature;
+        private readonly BufferingWriteStream _bufferStream;
+
+        public SendFileFeatureWrapper(IHttpSendFileFeature originalSendFileFeature, BufferingWriteStream bufferStream)
+        {
+            _originalSendFileFeature = originalSendFileFeature;
+            _bufferStream = bufferStream;
+        }
+
+        // Flush and disable the buffer if anyone tries to call the SendFile feature.
+        public Task SendFileAsync(string path, long offset, long? length, CancellationToken cancellation)
+        {
+            _bufferStream.DisableBuffering();
```

Comment:
Nice, I think we have some follow up items on MVC side where we have FileStreamResult, StreamOutputFormatter and we should do similar thing